### DISCUSSION TITLE
64-bit inode problem: _FILE_OFFSET_BITS=64 available for OpenWatcomv2?

### DISCUSSION THREAD

**Discussion by eddavis2** [open-watcom/open-watcom-v2#834] [Q&A] [answered]
Does Watcom v2 for Linux support _FILE_OFFSET_BITS=64 ?

Situation:
WLS v2, 64 bit Linux, in the /mnt/c/  native Windows file system.
Given a simple 32-bit app that uses readdir() and/or stat().  
I compile it with gcc -m32.  Using readdir() or stat(), both report "Value too large for defined data type" via perror(), when in the Windows file system at /mnt/c/ (works fine when I'm in /home, /bin, /tmp, /usr, and so on).
If I add #define _FILE_OFFSET_BITS 64 at the top of the source file, and recompile, magically it works.

Question:
Does Watcom v2 for Linux support _FILE_OFFSET_BITS=64 ?
My actual app requires Watcom C, but my app is currently stuck in 32-bit land.  Hence the question regarding _FILE_OFFSET_BITS=64, so I can navigate around the /mnt/c/ files.

See here - https://www.mjr19.org.uk/sw/inodes64.html - for a discussion of the problem.

Thank you.

**Comment by jmalak** [comment, marked as answer]
OW2 doesn't use _FILE_OFFSET_BITS macro because we use microsoft related 64-bit functions for I/O as stat64 etc.
Anyway OW doesn't support 64-bit targets only 32-bit targets.
Other thing is difference in long type size between Windows and Linux on 64-bit host
